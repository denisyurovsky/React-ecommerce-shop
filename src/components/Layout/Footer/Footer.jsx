import { Toolbar, Typography } from '@mui/material';
import React from 'react';

import styles from './Footer.module.scss';

export const Footer = () => {
  const date = new Date();

  return (
    <footer className={styles.appbar}>
      <Toolbar className={styles.toolBar}>
        <Typography
          data-testid="copyright"
          variant="body3"
          className={styles.copyright}
        >
          EPAM Systems Â© {date.getFullYear()}
        </Typography>
      </Toolbar>
    </footer>
  );
};
