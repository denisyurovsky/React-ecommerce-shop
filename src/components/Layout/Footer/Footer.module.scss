@import "../../../assets/styles/colors.scss";

.appbar {
  padding: 0;
  background-color: $greyMain;
  position: relative !important;
  height: min-content;
  width: 100%;
}

.copyright {
  height: min-content;
}

.toolBar {
  margin: 0;
  display: flex;
  flex-direction: row;
  justify-content: center !important;
}
