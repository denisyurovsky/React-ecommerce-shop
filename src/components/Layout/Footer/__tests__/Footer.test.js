import { render } from '@testing-library/react';
import React from 'react';

import { Footer } from '../Footer';
jest.useFakeTimers('modern');

describe('Footer component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot when year 2025', () => {
      jest.setSystemTime(new Date(2025, 1, 1));

      const { asFragment } = render(<Footer />);

      expect(asFragment()).toMatchSnapshot();
    });
    it('renders a valid snapshot when year 2021', () => {
      jest.setSystemTime(new Date(2021, 1, 1));

      const { asFragment } = render(<Footer />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
