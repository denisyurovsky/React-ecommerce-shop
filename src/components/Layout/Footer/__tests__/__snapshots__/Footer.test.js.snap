// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component snapshots renders a valid snapshot when year 2021 1`] = `
<DocumentFragment>
  <footer
    class="appbar"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolBar css-1t29gy6-MuiToolbar-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body3 copyright css-37jdci-MuiTypography-root"
        data-testid="copyright"
      >
        EPAM Systems © 2021
      </span>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer component snapshots renders a valid snapshot when year 2025 1`] = `
<DocumentFragment>
  <footer
    class="appbar"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolBar css-1t29gy6-MuiToolbar-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body3 copyright css-37jdci-MuiTypography-root"
        data-testid="copyright"
      >
        EPAM Systems © 2025
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
